src/background.js

```diff
@@ -28,6 +34,46 @@ browser.browserAction.onClicked.addListener(() => {
     openOverview()
 })
 
+
+browser.bookmarks.onCreated.addListener((id, bookmarkInfo) => {
+    const samePageCandidates = (await findPagesByUrl({url})).rows.map(row => row.doc)
+    if (samePageCandidates.length <= 0) {
+        const favIconBlob = await dataURLToBlob(pageData.favIconURI)
+        const freezeDryBlob = new Blob([pageData.freezeDryHTML], {type: 'text/html;charset=UTF-8'})
+        try {
+            const pageData = await fetchPageData(bookmarkInfo.url, {
+                                                    includeFreezeDry: true,
+                                                    includePageContent: true,
+                                                    includeFavIcon: true,
+                                                })
+            const pageDoc = {
+                url: bookmarkInfo.url,
+                title: bookmarkInfo.title,
+                content: pageData.content,
+                _id: generatePageDocId(bookmarkInfo.dateAdded, id),
+                _attachments: [{
+                        content_type: freezeDryBlob.type,
+                        data: freezeDryBlob
+                    }, {
+                        content_type: favIconBlob.type,
+                        data: favIconBlob
+                    }
+                ]
+            }
+        } catch (err) {
+            console.log("Error occurred while fetching page data: " + err.toString())
+        } finally {
```

Comment:
`finally` blocks happen regardless of whether an error is thrown in the `try` block, i.e., always, even if you return from `try`/`catch`. They're generally a good place to do stuff that you want to do regardless of outcome. So the `db.bulkDocs()` call might be good here and in the `try` and `catch` you just alter some `pageDoc` and `bookmarkDoc` variables that are declared in the outer scope?